sphinxcontrib/opencontracting.py

```diff
@@ -211,11 +213,104 @@ def run(self):
         return [admonition_node]
 
 
+class WorkedExampleListNode(nodes.General, nodes.Element):
+    """
+    And empty class only used for identifying the directive references through the documentation
+    """
+    pass
+
+
+class WorkedExampleList(Directive):
+    required_arguments = 1
+    final_argument_whitespace = True
+    option_spec = {'block': directives.unchanged}
+
+    def run(self):
+        title = self.arguments[0]
+        block = self.options.pop('block', '')
+        return [WorkedExampleListNode(block=block, title=title)]
+
+
+class WorkedExample(Directive):
+    required_arguments = 1
+    final_argument_whitespace = True
+    option_spec = {'block': directives.unchanged}
+
+    def run(self):
+        env = self.state.document.settings.env
+        title = self.arguments[0]
+        target_id = f'worked-example-{env.new_serialno("worked-example")}'
+        target_node = nodes.target('', '', ids=[target_id])
+        # A dummy node as we don't want to show anything for the worked example, only mark the content as one
+        ocds_block = self.options.pop('block', '')
+        node = nodes.paragraph('')
+        if not hasattr(env, WORKED_EXAMPLES_ENV_NAME):
+            setattr(env, WORKED_EXAMPLES_ENV_NAME, [])
+
+        getattr(env, WORKED_EXAMPLES_ENV_NAME).append({
+            'docname': env.docname,
+            'lineno': self.lineno,
+            'target': target_node,
+            'block': ocds_block,
+            'title': title,
+        })
+
+        return [target_node, node]
+
+
+def purge_worked_examples(app, env, docname):
+    """
+    Method for removing any existing worked examples from old builds
+    """
+    if not hasattr(env, WORKED_EXAMPLES_ENV_NAME):
+        return
+
+    setattr(env, WORKED_EXAMPLES_ENV_NAME, [worked_example for worked_example in getattr(env, WORKED_EXAMPLES_ENV_NAME)
+                                            if worked_example['docname'] != docname])
```

Comment:
Abbreviated to `example` since it's clear enough, and my earlier `WORKEDEXAMPLE_ENV_ATTRIBUTE` change would make the line too long.

```suggestion
    setattr(env, WORKEDEXAMPLE_ENV_ATTRIBUTE, [example for example in getattr(env, WORKEDEXAMPLE_ENV_ATTRIBUTE)
                                             if example['docname'] != docname])
```